packages/geoprocessing/src/metrics/helpers.ts

```diff
@@ -316,45 +316,78 @@ const classSortAlphaDisplay = (a: DataClass, b: DataClass) => {
+ * @param numerators array of metrics, to be used as numerators (often sketch metrics)
+ * @param denominators array of metrics, to be used as denominators (often planning region metrics)
+ * @param metricIdOverride optional metricId value to assign to outputted metrics
+ * @param idProperty optional id property to match metric with total metric, defaults to classId
+ * @returns Metric[] of percent values
  */
 export const toPercentMetric = (
-  metrics: Metric[],
-  totals: Metric[],
-  /** Set percent metrics with new metricId.  Defaults to leaving the same */
-  percMetricId?: string
+  numerators: Metric[],
+  denominators: Metric[],
+  options: {
+    metricIdOverride?: string;
+    idProperty?: string;
+  } = {}
 ): Metric[] => {
+  const { metricIdOverride, idProperty = "classId" } = options;
+
+  // Index denominators into precalc totals using idProperty
   const totalsByKey = (() => {
-    return keyBy(totals, (total) =>
-      total.classId ? total.classId : total.metricId
-    );
+    return keyBy(denominators, (total) => String(total[idProperty]));
   })();
-  return metrics.map((curMetric) => {
-    if (!curMetric || curMetric.value === undefined)
-      throw new Error(`Malformed metrics: ${JSON.stringify(curMetric)}`);
 
-    const idProperty = curMetric.classId ? "classId" : "metricId";
+  // For each metric in metric group
+  return numerators.map((numerMetric) => {
+    if (!numerMetric || numerMetric.value === undefined)
+      throw new Error(
+        `Malformed numerator metric: ${JSON.stringify(numerMetric)}`
+      );
 
-    const idValue = curMetric[idProperty];
-    if (!idValue) throw new Error(`Missing total index: ${idValue}`);
+    const idValue = numerMetric[idProperty];
 
-    const value = curMetric[idProperty];
-    if (!value)
+    if (idValue === null || idValue === undefined)
       throw new Error(
-        `Missing metric id property ${idProperty}, ${JSON.stringify(curMetric)}`
+        `Invalid ${idProperty} found in numerator metric: ${JSON.stringify(
+          numerMetric
+        )}`
       );
-    const totalMetric = totalsByKey[idValue];
-    if (!totalMetric) {
+
+    const denomMetric = totalsByKey[idValue];
+    if (!denomMetric) {
       throw new Error(
-        `Missing total: ${idProperty}: ${JSON.stringify(curMetric)}`
+        `Missing matching denominator metric with ${idProperty} of ${idValue} for numerator: ${JSON.stringify(
+          numerMetric
+        )}`
       );
     }
+    if (denomMetric.value === null || denomMetric.value === undefined) {
+      throw new Error(
+        `Malformed denominator metric: ${JSON.stringify(numerMetric)}`
+      );
+    }
+
+    const value = (() => {
+      // Catch 0 or malformed denominator value and return percent metric with 0 value
+      if (denomMetric.value === 0) {
+        console.log(
+          `Denominator metric with ${idProperty} of ${idValue} has 0 value, returning 0 percent metric`
+        );
+        return 0;
```

Comment:
In project space in the Azores, instead of returning 0 here, I return NaN.

Reply:
Perfect.  I am getting ClassTable to support the Info tooltop for non-chart layout as well 
<img width="157" alt="image" src="https://github.com/seasketch/geoprocessing/assets/81816/e7e1233e-d850-43bc-8025-5aa4af8253f9">